signature/plugin.go

```diff
@@ -190,7 +159,8 @@ func (s *pluginSigner) mergeConfig(config map[string]string) map[string]string {
 }
 
 func (s *pluginSigner) generateSignatureEnvelope(ctx context.Context, desc notation.Descriptor, opts notation.SignOptions) ([]byte, error) {
-	rawDesc, err := json.Marshal(desc)
+	payload := notation.Payload{TargetArtifact: desc}
+	rawDesc, err := json.Marshal(payload)
 	if err != nil {
 		return nil, err
```

Comment:
Should this be similar to line 106?

```suggestion
		return nil, fmt.Errorf("envelope payload can't be marshaled: %w", err)
```